penguin/llm/runtime.py

```
@@ -18,37 +26,70 @@
-    tools_getter = getattr(tool_manager, "get_responses_tools", None)
-    tools_payload = tools_getter() if callable(tools_getter) else []
+    tools_payload = _get_tool_payload(
+        tool_manager,
+        include_web_search=tool_format == "openai_responses",
+    )
     if not tools_payload:
         return extra_kwargs
 
-    if uses_openai_native_responses:
-        setattr(model_config, "interrupt_on_tool_call", True)
+    setattr(model_config, "interrupt_on_tool_call", True)
```

Ruff (0.15.11)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)